artemis-service-extensions/src/main/java/org/apache/activemq/artemis/service/extensions/xa/recovery/ActiveMQXAResourceWrapper.java

```diff
@@ -67,6 +67,15 @@ public ActiveMQXAResourceWrapper(XARecoveryConfig... xaRecoveryConfigs) {
       }
    }
 
+   public void updateRecoveryConfig(XARecoveryConfig... xaRecoveryConfigs) {
+      synchronized (ActiveMQXAResourceWrapper.lock) {
+         if (delegate != null) {
+            delegate = null;
```

Comment:
shouldn't close be called here instead? the previous resource will probably leak?


isn't there a way to write a test for this?

Reply:
of course you could have the close called with a try .. catch (Exception ignored) {}